static/js/components/EmploymentForm.js

```diff
@@ -65,15 +61,38 @@ class EmploymentForm extends ProfileFormFields {
   };
 
   openNewWorkHistoryForm: Function = (): void => {
-    openNewWorkHistoryForm.call(this);
+    const {
+      updateProfile,
+      profile,
+      setWorkDialogIndex,
+      setWorkDialogVisibility,
+      validator,
+    } = this.props;
+    let clone = {
+      ...profile,
+      work_history: [...profile.work_history, generateNewWorkHistory()]
+    };
+    updateProfile(clone, validator, true);
+    setWorkDialogIndex(clone.work_history.length - 1);
```

Comment:
sure, I just copy and pasted what was already there, happy to do a little refactoring now